src/hooks/sync-sites/use-sync-pull.ts

```diff
@@ -128,46 +137,108 @@ export function useSyncPull( {
 		[ __, client, pullStatesProgressInfo, updatePullState ]
 	);
 
+	const checkBackupFileSize = async ( downloadUrl: string ): Promise< number > => {
+		try {
+			return await getIpcApi().checkSyncBackupSize( downloadUrl );
+		} catch ( error ) {
+			console.log( 'Failed to check backup file size', error );
+			Sentry.captureException( error );
+			throw new Error( 'Failed to check backup file size' );
+		}
+	};
+
 	const onBackupCompleted = useCallback(
 		async ( remoteSiteId: number, backupState: SyncBackupState & { downloadUrl: string } ) => {
 			const { downloadUrl, selectedSite, isStaging } = backupState;
-			updatePullState( selectedSite.id, remoteSiteId, {
-				status: pullStatesProgressInfo.downloading,
-				downloadUrl,
-			} );
 
-			const filePath = await getIpcApi().downloadSyncBackup( remoteSiteId, downloadUrl );
+			try {
+				const fileSize = await checkBackupFileSize( downloadUrl );
+				console.log( 'Backup file size:', { fileSize, limit: SYNC_PUSH_SIZE_LIMIT_BYTES } );
 
-			updatePullState( selectedSite.id, remoteSiteId, {
-				status: pullStatesProgressInfo.importing,
-			} );
+				if ( fileSize > SYNC_PUSH_SIZE_LIMIT_BYTES ) {
+					console.log( 'File size exceeds limit, prompting user' );
+					const { response: userChoice } = await getIpcApi().showMessageBox( {
+						type: 'warning',
+						title: __( 'Large Backup File' ),
+						message: sprintf(
+							__(
+								'The backup file is over %s GB. You will not be able to push back the site. Continue?'
```

Comment:
We could improve the message wording, e.g.:

```suggestion
								"Your site's backup exceeds %s GB. Pulling it will prevent you from pushing the site back. Do you want to continue?"
```

Could we also add a line break before the question and change title/message fields to message/details, similarly to 'Overwrite Production site'?

Reply:
Re: line break, sure!

Re: copy, it's not the pulling that prevents it, so maybe we shouldn't emphasize it? How about this:

`Your site's backup exceeds %s GB. You will not be able to push the site back. Do you want to continue?`

Reply:
I think the version I proposed is more precise. If the Studio site currently takes 50MB, and the user is going to pull a 2500 MB site, the pulling is the exact reason that will prevent them from pushing it back.

Reply:
Ah yes, makes sense. Thanks for explaining!

Reply:
Could you please adjust how we display the message, mentioned in my previous comment?

> Could we also add a line break before the question and change title/message fields to message/details, similarly to 'Overwrite Production site'?

Reply:
Sorry missed doing that, done now!